packages/core/src/helpers/logger/device-console.logger.ts

```diff
@@ -1,42 +1,54 @@
-import { AbstractLogger, LogLevel, UnknownLogLevelError } from './logger';
+import { Logger, LogLevel, UnknownLogLevelError } from './logger';
 
-export class DeviceConsoleLogger extends AbstractLogger {
-    protected logger: Console;
+export class DeviceConsoleLogger extends Logger {
+    private logger: Console;
+    private tag?: string;
 
-    constructor(logger?: Console) {
+    constructor(logger?: Console, tag?: string) {
         super();
-        this.logger = logger || console;
+        this.logger = logger ?? console;
+        this.tag = tag;
     }
 
-    logKeyValue(key: string, value: unknown): void {
+    public logKeyValue(key: string, value: unknown): void {
         this.info(`${key}=${value}`);
     }
 
-    log(level: LogLevel, message: string): void;
-    log(level: LogLevel, tag: string, message: string): void;
-    log(level: LogLevel, tagOrMessage: string, message?: string): void {
-        if (message) {
-            message = `[${tagOrMessage}] ${message}`;
-        } else {
-            message = tagOrMessage;
+    protected createLoggerWithTag(tag: string): Logger {
+        return new DeviceConsoleLogger(this.logger, tag);
+    }
+
+    protected handleLog(level: LogLevel, parameters: unknown[]): void {
+        parameters = parameters ?? [];
+
+        if (this.tag) {
+            const tagStr = `[${this.tag}]`;
+
+            if (parameters.length === 0 || typeof parameters[0] !== 'string') {
+                // Set tag as first parameters array element
+                parameters.unshift(tagStr);
+            } else {
+                // Mix tag with parameter as it might have formatting placeholders
+                parameters[0] = `${tagStr} ${parameters[0]}`;
```

Comment:
Could you implement a simple example of the usage of this method? I'm not sure about this line, maybe it's that I'm not understanding all the possible "elses". I think that unknown is sometimes too imprecise.

Reply:
I've added some extra documentation, let me know if now it's clearer the purpose of these branches: https://github.com/mobilejazz/harmony-typescript/pull/124/commits/88fe70e82525492090e2e10d5bbf24b8376f95c0

The difference is:
- Mixed types: `console.log(true, 42, { foo: 'bar' })`
- Template: `console.log('The meaning of life is %s', 42)`

In the second case, if we just prepend the tag to the parameters array the template won't work:

![image](https://user-images.githubusercontent.com/188612/207016659-ba42eea6-cf47-453b-8269-a63b1b9c6949.png)

Reply:
Nice! Thanks for the answer